### ISSUE TITLE
Migrate to rules_js

### ISSUE THREAD

**Issue by gonzojive** [gTile/gTile#308]
[rules_js](https://github.com/aspect-build/rules_js) is better supported than [rules_nodejs](https://github.com/bazelbuild/rules_nodejs). It might be good to try migrating to that. I don't have time for this, but someone else might.

**Comment by scherepanov** [comment]
I did not realised that this issue actually prevents build and tests to run.
Looking on migration... require learning bazel... may be someone who works with bazel will help.

**Comment by gonzojive** [comment]
I don't think this prevents building.

**Comment by scherepanov** [comment]
It is preventing build:

```
ERROR: Traceback (most recent call last):
        File "/home/user/git/public/gTile/WORKSPACE", line 1, column 10, in <toplevel>
                workspace(
Error in workspace: workspace() got unexpected keyword argument 'managed_directories'
ERROR: Error computing the main repository mapping: Encountered error while reading extension file 'tools/build_defs/repo/http.bzl': no such package '@bazel_tools//tools/build_defs/repo': error loading package 'external': Could not load //external package
```
I have on my box same error as in #314

**Comment by gonzojive** [comment]
Using bazel 5 and bazelisk fixes it.

However a migration should happen. I did most of what's needed in my bazel6
branch.

The main nontrivial work left is getting the transpiled JavaScript to be
compatible with gjs and its quirks. That will take some brushing up on how
rolllup and tsc work in combination to transpile the typescript.

If someone knows of another gnome extension it would be helpful to look at
their build process. Thanks for any help on that.

**Comment by gonzojive** [comment]
I think the way it should work is that tsc should transpile with
target=es2015, module=es6. Roll-up should then consume all of the js files
(es6 modules) and output "iife" (self executable format).

**Comment by scherepanov** [comment]
Sorry, I am quite far from Bazel builds, not really understand what has to be done.

**Comment by gonzojive** [comment]
Try testing my bazel6 branch. I think it is all fixed and working but
haven't tested it.

**Comment by scherepanov** [comment]
Build works with bazel 6!
Please open PR, I will merge.

**Comment by scherepanov** [comment]
Build works but produces non working code, metadata.json is missing (at
least)

**Comment by scherepanov** [comment]
Checking more... :install-extension leaves empty gTile dir, no files installed.

**Comment by gonzojive** [comment]
Here is the PR. Thanks for testing.

https://github.com/gTile/gTile/pull/318

I had commented out the line in the install script that actually performed the unzip. Check out the latest version of the `bazel6-pr` branch to get those changes.

It might be worth inspecting the contents of bazel-bin/dist.tar.gz to look for any possible problematic diffs between the old version and the version after the PR.

**Comment by scherepanov** [comment]
Fixed in #318
